docs/user_guides/fs/storage_connector/usage.md

```diff
@@ -1,9 +1,138 @@
 # Storage Connector Usage
+Here, we look at how to use a storage connector after its created. 
+Storage connectors provide an important first step of integrating to an external source.
+The 3 fundamental functionalities where storage connectors are used are
+    
+1. Reading data into spark dataframe
+2. Creating an external feature group
+3. Writing training data
+
+We look at each case in below sections. 
 
 ## Retrieving a Storage Connector
+We retrieve a storage connector simply by its unique name.
+
+=== "PySpark"
+    ```python
+    import hsfs
+    # Connect to the Hopsworks feature store
+    hsfs_connection = hsfs.connection()
+    # Retrieve the metadata handle
+    feature_store = hsfs_connection.get_feature_store()
+    # Retrieve storage connector
+    connector = feature_store.get_storage_connector('connector_name')
+    ```
+
+=== "Scala"
+    ```scala
+    import com.logicalclocks.hsfs._
+    val connection = HopsworksConnection.builder().build();
+    val featureStore = connection.getFeatureStore();
+    // get directly via connector sub-type class e.g. for GCS type
+    val connector = featureStore.getGcsConnector("gcs_test")
+    ```
 
 ## Reading a Spark Dataframe from a Storage Connector
 
+One of the most common usage of a storage connector is to read data directly into a spark dataframe.
+Its achieved via the `read` API of the connector object, which hides all the complexity of authentication and integration 
+with a data storage source. 
+The `read` API primarily has two parameters for specifying the data source, `path` and `query`, depending on the storage connector type.
+The exact behaviour could change depending on the storage connector type, but broadly they could be classified as below
+### Data lake/object based connectors
+
+For data sources based on object/file storage such as AWS S3, ADLS, GCS, we set the full object path to the `path` argument
+and should pass a spark data format to `data_format` argument.
+
+=== "PySpark"
+    ```python
+    # read data into dataframe using path 
+    df = connector.read(data_format='csv',path='path/to/object')   
+    ```
+
+=== "Scala"
+    ```scala 
+    // read data into dataframe using path     
+    val df = connector.read("", "csv" , new HashMap(),"gs://bucket/test/")  
+    ```
```

Comment:
```suggestion
For data sources based on object/file storage such as AWS S3, ADLS, GCS, we set the full object path in the `path` argument
and users should pass a Spark data format to the `data_format` argument.

=== "PySpark"
    ```python
    # read data into dataframe using path 
    df = connector.read(data_format='csv', path='path/to/object')   
    ```

=== "Scala"
    ```scala 
    // read data into dataframe using path     
    val df = connector.read("", "csv", new HashMap(), "gs://bucket/test/")  
    ```
```